bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/net/CidrAddress.java

```diff
@@ -29,11 +35,71 @@
     private final InetAddress address;
     private final int prefix;
 
-    public CidrAddress(@NonNull InetAddress address, short networkPrefixLength) {
+    private static final Pattern IPV4PATTERN = Pattern
+            .compile("^(([01]?\\d\\d?|2[0-4]\\d|25[0-5])\\.){3}([01]?\\d\\d?|2[0-4]\\d|25[0-5])$");
+    private static final Pattern IPV6PATTERN = Pattern
+            .compile("^(([0-9a-fA-F]{1,4}:){7}([0-9a-fA-F]){1,4}(%[\\S]*)?)$");
+
+    public CidrAddress(@NonNull InetAddress address, int networkPrefixLength) {
         this.address = address;
+        checkPrefix(networkPrefixLength);
         this.prefix = networkPrefixLength;
     }
 
+    /**
+     * Return true if the given string is a valid IPv4 or IPv6 address.
+     *
+     * Be aware that only full IP addresses (4 segments for IPv4 and 6 segments for IPv6)
+     * are considered as correct input.
+     * For instance "::" although a valid IPv6 address per specification, will
+     * not be recognised as valid by this method.
+     *
+     * @param ipAddress An address like 129.12.3.1 or "2001:DB8:0:0:8:800:200C:417A%eth0"
+     * @return
+     */
+    public static boolean isIpAddress(String ipAddress) {
+        Matcher m1 = IPV4PATTERN.matcher(ipAddress);
+        if (m1.matches()) {
+            return true;
+        }
+        Matcher m2 = IPV6PATTERN.matcher(ipAddress);
+        return m2.matches();
+    }
+
+    public CidrAddress(String cidrNotation) throws IllegalArgumentException {
+        if (StringUtils.isBlank(cidrNotation)) {
+            throw new IllegalArgumentException("cidrNotation cannot be null or empty");
+        }
+        String[] parts = cidrNotation.split("/");
+
+        if (!isIpAddress(parts[0])) {
+            throw new IllegalArgumentException("IP address is not valid");
+        }
+        try {
+            address = InetAddress.getByName(parts[0]);
+        } catch (UnknownHostException e) {
+            throw new IllegalArgumentException("IP address is not valid");
+        }
+
+        if (parts.length == 2) {
+            prefix = Integer.valueOf(parts[1]);
+            checkPrefix(prefix);
+        } else if (parts.length == 1) {
+            prefix = 0;
+        } else {
+            throw new IllegalArgumentException("Not a valid CIDR notation");
+        }
+    }
+
+    private void checkPrefix(int prefix) throws IllegalArgumentException {
+        if (prefix % 2 != 0) {
+            throw new IllegalArgumentException("Not a valid prefix. This must be valid: prefix mod 2 == 0");
+        }
+        if (prefix < 0 || prefix >= address.getAddress().length * 8) {
```

Comment:
One error that I also have done in my previous PR: [RFC1878](https://tools.ietf.org/html/rfc1878) allows a prefix of `32` for a single host route, so you should test `prefix > address.getAddress().length * 8`

Reply:
@davidgraeff Did you see this comment regarding the `32` bit netmask?